### PR TITLE
Add pg_repack as default maintenance strategy

### PR DESCRIPTION
**Pull request by npwalker**

### CODE REVIEW

File: manifests/maintenance/pg_repack.pp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    hour     => 5,
+    minute   => 30,
+    command  => "${repack} ${other_tables} ${logging}",
+    require  => File[$logging_directory],
+  }
+
+  cron { 'pg_repack reports tables' :
+    ensure   => $ensure_cron,
+    user     => 'root',
+    weekday => [3],
```
Comment (npwalker) [inline manifests/maintenance/pg_repack.pp]: Is this too frequent?

Comment (npwalker) [reply manifests/maintenance/pg_repack.pp]: going to go with once a month to start instead

File: manifests/maintenance/pg_repack.pp [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    true    => absent,
+    default => present
+  }
+
+  $repack          = 'su - pe-postgres -s /bin/bash -c "/opt/puppetlabs/server/apps/postgresql/bin/pg_repack -d pe-puppetdb'
+  $facts_tables    = '-t factsets -t fact_paths"'
+  $catalogs_tables = '-t catalogs -t catalog_resources -t edges -t certnames"'
+  $other_tables    = '-t producers -t resource_params -t resource_params_cache"'
+  $reports_tables  = '-t reports"'
+  $logging         = "> ${logging_directory}/output.log 2>&1"
```
Comment (reidmv) [inline manifests/maintenance/pg_repack.pp]: Is there any logic to ensure that multiple repacks cannot be started simultaneously? That's the biggest danger of automated repacking.

File: manifests/maintenance.pp [hunk 3]
```diff
@@ -1,79 +1,25 @@
...
-  Optional[Integer] $maint_cron_minute  = undef, #DEPRECATED
+  Boolean $disable_maintenance = false,
   String  $logging_directory  = '/var/log/puppetlabs/pe_databases_cron',
   String  $script_directory   = $pe_databases::scripts_dir,
 ){
 
-  $ensure_cron = $disable_maintenace ? {
-    true    => absent,
-    default => present
+  #If the PE Version includes pg_repack (2018.1.7 and 2019.0.2) then use pg_repack and remove the old script and cron jobs
```
[the remark this answers is not in the document]
Comment (tkishel) [reply manifests/maintenance.pp]: Untested:

```
Facter.add(:pg_repack) do
  confine Facter.value(:pe_postgresql_info)['installed_server_version'] => '9.6'
  setcode do
    cmd = %q(su - pe-postgres -s /bin/bash -c "/opt/puppetlabs/server/bin/psql -d pe-puppetdb -c 'SELECT * FROM pg_extension'")
    res = Facter::Util::Resolution.exec(cmd)
    res.include?('pg_repack')
  end
end
```

File: manifests/maintenance/pg_repack.pp (hunk 2 above)
Comment (npwalker) [reply manifests/maintenance/pg_repack.pp]: The only ones that could overlap are the other tables and the reports table which run at 5:30 on 1 day of the month while facts and catalogs run at 4:30 2 days a week.  

I'm not particularly concerned about them overlapping... but if we thought it was a serious concern then we could move the times they run further apart.

File: manifests/maintenance.pp (hunk 3 above)
[the remark this answers is not in the document]
Comment (npwalker) [reply manifests/maintenance.pp]: if someone really wants to use pg_repack on an older version of PE they can set manage_database_maintenance to false and then include the pg_repack manifest via a profile.

I don't suspect the use of manually installed pg_repack will be high in older installs.

**Pull request merged**